filters/LiTreeFilter.cpp

```diff
@@ -232,16 +220,31 @@ void LiTreeFilter::segmentTree(PointView& view, PointIdList& Ui,
 
     // "We then insert a dummy point n0 that is far away (e.g., 100m) from t0
     // into Ni."
+    //
+    // If the found dummy point is the target point, then the target point is
+    // geographically isolated. We remove it and return.
     PointId n0 = locateDummyPoint(view, Ui, t0);
+    if (n0 == t0)
+    {
+        Utils::remove(Ui, t0);
+        return;
+    }
     Ni.push_back(n0);
 
     // "For iteration i, we classify each point in Ui as Pi or Ni."
     double tx = view.getFieldAs<double>(Id::X, t0);
     double ty = view.getFieldAs<double>(Id::Y, t0);
     for (PointId const& ui : Ui)
     {
+        // t0 and n0 are already in Pi or Ni.
+        if (ui == t0 || ui == n0)
+            continue;
+
         double ux = view.getFieldAs<double>(Id::X, ui);
         double uy = view.getFieldAs<double>(Id::Y, ui);
+        //ABELL - This is strange. We've already done a radius search to locate a dummy point
+        //  and the distance used is 100, so we already have the points that are within that
+        //  distance, so there's really no need to do it again, is there?
```

Comment:
I agree that it's clunky. This was inserted to speed computation. It would still have some benefit if the radius for the dummy point were larger than 100m.